plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/AbstractASTManager.java

```diff
@@ -2235,30 +2235,41 @@ public static String getTokToSearchInOtherModule(Tuple3<IModule, String, IToken>
             return ret;
         }
 
-        if (!isAbsoluteImportEnabledx) {
-            //ok, one last shot, to see a relative looking in folders __init__
-            modTok = findModuleFromPath(asRelativeImport, nature, false, null);
-            mod = modTok.o1;
-            if (checkValidity(currentModuleName, mod, true)) {
-                Tuple<IModule, String> ret = fixTok(modTok, tok, activationToken);
-                //now let's see if what we did when we found it as a relative import is correct:
-
-                //if we didn't find it in an __init__ module, all should be ok
-                if (!mod.getName().endsWith("__init__")) {
-                    return ret;
-                }
-                //otherwise, we have to be more cautious...
-                //if the activation token is empty, then it is the module we were looking for
-                //if it is not the initial token we were looking for, it is correct
-                //if it is in the global tokens of the found module it is correct
-                //if none of this situations was found, we probably just found the same token we had when we started (unless I'm mistaken...)
-                else if (activationToken.length() == 0 || ret.o2.equals(activationToken) == false
-                        || mod.isInGlobalTokens(activationToken, nature, false, state)) {
+        if (isAbsoluteImportEnabledx) {
```

Comment:
I believe that the change in this file should be reverted.

I think this only happened because you were setting the grammar to python 3 on the test and it failed and it wasn't restoring properly (with the latest version where the `try..finally`is done this shouldn't happen anymore).

With this change relative imports are being checked when only absolute imports are allowed, which is not correct.